server/schedulers/balance_region.go

```diff
@@ -232,25 +235,27 @@ func (s *balanceRegionScheduler) Schedule(cluster schedule.Cluster, dryRun bool)
 }
 
 // transferPeer selects the best store to create a new peer to replace the old peer.
-func (s *balanceRegionScheduler) transferPeer(solver *solver, collector *plan.Collector) *operator.Operator {
+func (s *balanceRegionScheduler) transferPeer(solver *solver, collector *plan.Collector, dstStores []*core.StoreInfo, faultStores []*core.StoreInfo) *operator.Operator {
+	excludeTargets := solver.region.GetStoreIDs()
+	for _, store := range faultStores {
+		excludeTargets[store.GetID()] = struct{}{}
+	}
 	// the order of the filters should be sorted by the cost of the cpu overhead.
 	// the more expensive the filter is, the later it should be placed.
 	filters := []filter.Filter{
-		filter.NewExcludedFilter(s.GetName(), nil, solver.region.GetStoreIDs()),
-		filter.NewSpecialUseFilter(s.GetName()),
-		&filter.StoreStateFilter{ActionScope: s.GetName(), MoveRegion: true},
-		filter.NewRegionScoreFilter(s.GetName(), solver.source, solver.GetOpts()),
+		filter.NewExcludedFilter(s.GetName(), nil, excludeTargets),
 		filter.NewPlacementSafeguard(s.GetName(), solver.GetOpts(), solver.GetBasicCluster(), solver.GetRuleManager(), solver.region, solver.source),
 	}
 
-	candidates := filter.NewCandidates(solver.GetStores()).
-		FilterTarget(solver.GetOpts(), collector, filters...).
-		Sort(filter.RegionScoreComparer(solver.GetOpts()))
+	candidates := filter.NewCandidates(dstStores).FilterTarget(solver.GetOpts(), collector, filters...)
```

Comment:
Will it affect the current implementation of diagnostic?

Reply:
yes, I will record the reason why the store can not be selected and put it into the collector.